beam/src/components/BeamFilter.vue

```diff
@@ -0,0 +1,71 @@
+<template>
+	<div ref="beam-filters" class="beam_filters" :style="{ height: isOpen ? '100%' : headerHeight }">
+		<div ref="beam-filters-header" @click="toggle" class="beam_filters-heading">
+			<ToggleArrow :open="isOpen" />
+			<BeamHeading> Filter </BeamHeading>
+		</div>
+		<div class="beam_filters-options">
+			<slot>
+				<p>OPTIONS GO HERE</p>
+			</slot>
+		</div>
+	</div>
+</template>
+<script setup lang="ts">
+import BeamHeading from './BeamHeading.vue'
+import { ref, onMounted, useTemplateRef } from 'vue'
+
+let isOpen = ref(false)
+const header = useTemplateRef('beam-filters-header')
+const beamFilters = useTemplateRef('beam-filters')
+
+let totalHeight = ref(null)
+let headerHeight = ref(null)
+
+const toggle = () => {
+	isOpen.value = !isOpen.value
+}
+
+onMounted(() => {
+	headerHeight = getTotalHeight(header.value)
+	totalHeight = getTotalHeight(beamFilters.value)
+	beamFilters.value.style.height = headerHeight
```

Comment:
The `ref` variables will need to be get/set reactively:

```suggestion
	headerHeight.value = getTotalHeight(header.value)
	totalHeight.value = getTotalHeight(beamFilters.value)
	beamFilters.value.style.height = headerHeight.value
```